docs/migration/cosmoshub-4-v10-upgrade.md

```diff
@@ -0,0 +1,295 @@
+
+### Testing
+
+For those validator and full node operators that are interested in ensuring preparedness for the impending upgrade, you can run a [v10 Local Testnet](https://github.com/cosmos/testnets/tree/master/local) or join in our [Cosmos Hub Public Testnet](https://github.com/cosmos/testnets/tree/master/public).
+
+### Current runtime, cosmoshub-4 (pre-v10 upgrade) is running Gaia v9.1.1
+
+The Cosmos Hub mainnet network, `cosmoshub-4`, is currently running [Gaia v9.1.1](https://github.com/cosmos/gaia/releases/v9.1.1). We anticipate that operators who are running on v9.1.1, will be able to upgrade successfully. Validators are expected to ensure that their systems are up to date and capable of performing the upgrade. This includes running the correct binary, or if building from source, building with go `1.20`.
+
+### Target runtime, cosmoshub-4 (post-v10 upgrade) will run Gaia v10.0.0
+
+The Cosmos Hub mainnet network, `cosmoshub-4`, will run [Gaia v10.0.0](https://github.com/cosmos/gaia/releases/tag/v10.0.0). Operators _MUST_ use this version post-upgrade to remain connected to the network.
+
+## Upgrade steps
+
+There are 2 major ways to upgrade a node:
+
+- Manual upgrade
+- Upgrade using [Cosmovisor](https://github.com/cosmos/cosmos-sdk/tree/master/cosmovisor)
+    - Either by manually preparing the new binary
+    - Or by using the auto-download functionality (this is not yet recommended)
+
+If you prefer to use Cosmovisor to upgrade, some preparation work is needed before upgrade.
+
+### Method I: Manual Upgrade
+
+Make sure Gaia v10.0.0 is installed by either downloading a [compatable binary](https://github.com/cosmos/gaia/releases/tag/v10.0.0), or building from source. Building from source requires **Golang 1.20**.
+
+Run Gaia v9.1.1 till upgrade height, the node will panic:
+
+```shell
+ERR UPGRADE "v10" NEEDED at height: 15816200: upgrade to v10 and applying upgrade "v10" at height:15816200
+```
+
+Stop the node, and switch the binary to Gaia v10.0.0 and re-start by `gaiad start`.
+
+It may take several minutes to a few hours until validators with a total sum voting power > 2/3 to complete their node upgrades. After that, the chain can continue to produce blocks.
+
+### Method II: Upgrade using Cosmovisor
+
+::: warning
+<span style="color:red">**Please Read Before Proceeding**</span><br>
+Using Cosmovisor 1.2.0 and higher requires a lowercase naming convention for upgrade version directory. For Cosmovisor 1.1.0 and earlier, the upgrade version is not lowercased.
+:::
+
+> **For Example:** <br>
+> **Cosmovisor =< `1.1.0`: `/upgrades/v9-Lambda/bin/gaiad`**<br>
+> **Cosmovisor >= `1.2.0`: `/upgrades/v9-lambda/bin/gaiad`**<br>
+
+| Cosmovisor Version | Binary Name in Path |
+|--------------------|---------------------|
+| 1.3                | v10                 |
+| 1.2                | v10                 |
+| 1.1                | v10                 |
+| 1.0                | v10                 |
```

Comment:
This could be removed for v11-, even for v10 the Cosmovisor version will not matter (unless there are plans to bring back version names).

Reply:
I'd remove for v11 onwards with a small about cosmovisor behaviour.